engine/schema/src/com/cloud/usage/dao/UsageDaoImpl.java

```diff
@@ -469,4 +478,25 @@ public void removeOldUsageRecords(int days) {
             txn.close();
         }
     }
+
+    @SuppressWarnings("deprecation")
+    public Pair<List<? extends UsageVO>, Integer> getUsageRecordsPendingQuotaAggregation(final long accountId, final long domainId) {
+        final short opendb = TransactionLegacy.currentTxn().getDatabaseId();
+        TransactionLegacy.open(TransactionLegacy.USAGE_DB).close();
```

Comment:
@jburwell the dao uses connection multiplexing; so if we don't switch back to the previous DB several queries will fail as they will be execute on USAGE_DB instead of CLOUD_DB. This pattern is widely used across the codebase.